x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -681,13 +710,15 @@ AutoscalingDeciderResult scaleUpFromZero(List<String> waitingAnomalyJobs,
                                                        List<NodeLoad> nodeLoads,
                                                        List<String> waitingAnomalyJobs,
                                                        List<String> waitingAnalyticsJobs,
+                                                       List<String> waitingAllocatedModels,
                                                        @Nullable NativeMemoryCapacity futureFreedCapacity,
                                                        NativeMemoryCapacity currentScale,
                                                        MlScalingReason.Builder reasonBuilder) {
 
         // Are we in breach of maximum waiting jobs?
         if (waitingAnalyticsJobs.size() > numAnalyticsJobsInQueue
-            || waitingAnomalyJobs.size() > numAnomalyJobsInQueue) {
+            || waitingAnomalyJobs.size() > numAnomalyJobsInQueue
+            || waitingAllocatedModels.size() > 0) {
```

Comment:
Why the limit for models is `0` whereas for jobs it is some configurable number?

Reply:
@przemekwitek there is no reason to have a waiting model deployment, yet. If we discover that we need that setting, adding it later is not a big deal.